src/machi_lifecycle_mgr.erl

```diff
@@ -0,0 +1,1079 @@
+    ok;
+perhaps_bootstrap_chains([CD|ChainDefs], LocalFLUs_at_zero, LocalFLUs) ->
+    #chain_def_v1{full=Full, witnesses=Witnesses} = CD,
+    AllNames = [Name || #p_srvr{name=Name} <- Full ++ Witnesses],
+    case ordsets:intersection(ordsets:from_list(AllNames),
+                              ordsets:from_list(LocalFLUs_at_zero)) of
+        [] ->
+            perhaps_bootstrap_chains(ChainDefs, LocalFLUs_at_zero, LocalFLUs);
+        [FLU1|_]=FLUs ->
+            %% One FLU is enough: Humming Consensus will config the remaining
+            _ = bootstrap_chain(CD, FLU1),
+            perhaps_bootstrap_chains(ChainDefs, LocalFLUs_at_zero -- FLUs,
+                                    LocalFLUs -- FLUs)
+    end.
+
+bootstrap_chain(CD, FLU) ->
+    bootstrap_chain2(CD, FLU, 20).
+
+bootstrap_chain2(CD, FLU, 0) ->
+    lager:warning("Failed all attempts to bootstrap chain ~w via FLU ~w ",
+                  [CD,FLU]),
+    failed;
+bootstrap_chain2(#chain_def_v1{name=NewChainName, mode=NewCMode,
+                               full=Full, witnesses=Witnesses,
+                               old_full=ReqOldFull,
+                               old_witnesses=ReqOldWitnesses,
+                               props=_Props}=CD,
+                 FLU, N) ->
+    All_p_srvrs = Witnesses ++ Full,
+    L = [{Name, P_srvr} || #p_srvr{name=Name}=P_srvr <- All_p_srvrs],
+    MembersDict = orddict:from_list(L),
+    NewAll_list = [Name || #p_srvr{name=Name} <- All_p_srvrs],
+    NewWitnesses_list = [Name || #p_srvr{name=Name} <- Witnesses],
+
+    Mgr = machi_chain_manager1:make_chmgr_regname(FLU),
+    PStore = machi_flu1:make_projection_server_regname(FLU),
+    {ok, #projection_v1{epoch_number=OldEpoch, chain_name=OldChainName,
+                        mode=OldCMode,
+                        all_members=OldAll_list, witnesses=OldWitnesses}} =
+        machi_projection_store:read_latest_projection(PStore, private),
+    case set_chain_members(OldChainName, NewChainName, OldCMode,
+                           ReqOldFull, ReqOldWitnesses,
+                           OldAll_list, OldWitnesses,
+                           NewAll_list, Witnesses,
+                           Mgr, NewChainName, OldEpoch,
+                           NewCMode, MembersDict, NewWitnesses_list) of
+        ok ->
+            lager:info("Configured chain ~w via FLU ~w to "
+                       "mode=~w all=~w witnesses=~w\n",
+                       [NewChainName, FLU, NewCMode,
+                        NewAll_list, NewWitnesses_list]),
+            ok;
+        chain_bad_state=Else ->
+            lager:error("Attempt to bootstrap chain ~w via FLU ~w "
+                        "failed (no retries): ~w (defn ~w)\n",
+                        [NewChainName, FLU, Else, CD]),
+            Else;
+        Else ->
+            lager:error("Attempt to bootstrap chain ~w via FLU ~w "
+                        "failed: ~w (defn ~w)\n",
+                        [NewChainName, FLU, Else, CD]),
```

Comment:
I would like to see an attempt number in this error message, so its clear to a human that the procedure will be retried again.